app/Http/Controllers/Multiplayer/RoomsController.php

```diff
@@ -26,6 +33,100 @@ public function destroy($id)
         return response(null, 204);
     }
 
+    public function events($id)
+    {
+        $room = Room::findOrFail($id);
+
+        if (!$room->isRealtime()) {
+            throw new InvariantException('retrieving events is only supported for realtime rooms');
+        }
+
+        $params = get_params(request()->all(), null, [
+            'limit:int',
+            'after:int',
+            'before:int',
+        ], ['null_missing' => true]);
+        $after = $params['after'];
+        $before = $params['before'];
+        $limit = \Number::clamp($params['limit'] ?? 100, 1, 101);
```

Comment:
can use `$params[...]` directly?

Reply:
I'm not sure what you're asking of me here. The intention was to clamp `limit` to a sane range. I don't think `get_params()` does anything of that sort? Additionally `null_missing` is specified on the `get_params()` call, so in the worst case that *has* to be filled out with some default...?